auditwheel/repair.py

```diff
@@ -65,25 +65,43 @@ def repair_wheel(wheel_path: str, abi: str, lib_sdir: str, out_dir:
     return ctx.out_wheel
 
 
-def copylib(src_path, dest_path):
+def copylib(src_path, dest_dir):
+    """Graft a shared library from the system into the wheel and update the relevant links.
+
+    1) Copy the file from src_path to dest_dir/
+    2) Rename the shared object from soname to soname.<unique>
+    3) If the library has a RUNPATH/RPATH, update that to point to its new location.
+    """
     # Copy the a shared library from the system (src_path) into the wheel
     # if the library has a RUNPATH/RPATH to it's current location on the
     # system, we also update that to point to its new location.
 
-    print('Grafting: %s' % src_path)
+    with open(src_path, 'rb') as f:
+        shorthash = hashfile(f)[:8]
+
+    old_soname = elf_read_soname(src_path)
+    new_soname = '%s.%s' % (os.path.split(src_path)[1], shorthash)
```

Comment:
Maybe we could even make the naming scheme idempotent so that calling repair twice on a manylinux wheel would not change it. For instance (untested):

```python
base, ext = os.path.basename(src_path).split('.', 1)
if not base.endswith('-%s' % shorthash):
    new_soname = '%s-%s.%s' % (base, shorthash, ext)
else:
    new_soname = '%s.%s' % (base, ext)
```